### ISSUE TITLE
Deprecated item strikethrough stays behind until save

### ISSUE THREAD

**Issue by lnicola** [rust-analyzer/rust-analyzer#2195]
Before:

![image](https://user-images.githubusercontent.com/308347/68489643-2f27d400-0250-11ea-9761-6e8260ddfd5a.png)

After:

![image](https://user-images.githubusercontent.com/308347/68489660-364ee200-0250-11ea-8e5d-fb6664c68baa.png)

**Comment by kjeremy** [comment]
Are those coming from `cargo watch`?

**Comment by lnicola** [comment]
https://github.com/rust-analyzer/rust-analyzer/pull/1931, I assume.

**Comment by marcogroppo** [comment]
Isn't this true for all `cargo watch`-derived diagnostics? In your example if you replace `try!` with `nonexisting!` you get an error and it won't go away until you trigger `cargo watch` again

**Comment by lnicola** [comment]
This issue happens with `cargo watch` disabled, so I'm not sure why it's relevant. Anyway, the markers from `cargo watch` errors go away when I delete the code. Did you see my screenshot?

**Comment by marcogroppo** [comment]
sorry, I don't understand. If I write `try!` (or `"".trim_right()`) I don't get the "deprecated item" strikethrough unless I save the file and I have `cargo watch` enabled. If I do enable `cargo watch` and I write `nonexisting!(write!(&mut s, "{}", 1))` (like in your screenshot, just with a non-existing macro) I get an error diagnostic and I can see the issue I think you're observing (unless I've misunderstood you).

![image](https://user-images.githubusercontent.com/2735511/68534347-e8130f00-0333-11ea-9d28-0863700e89b7.png)

**Comment by lnicola** [comment]
> I don't get the "deprecated item" strikethrough unless I save the file and I have cargo watch enabled

Sorry, I think you're right. I tried to disable it, but I think I needed to reload the extension for it to apply. Okay, it's enabled and I'm on the 2015 edition.

There is a slight difference, though:

![image](https://user-images.githubusercontent.com/308347/68534766-f4ea2f80-0340-11ea-808d-1a1a21e1445e.png)

If I select `xxx` and type `write` without saving, I now see:

![image](https://user-images.githubusercontent.com/308347/68534775-13502b00-0341-11ea-8de1-6a6c5290eb06.png)

But with `try!`:

![image](https://user-images.githubusercontent.com/308347/68534778-1d722980-0341-11ea-8c33-5756ca6cc2a9.png)

![image](https://user-images.githubusercontent.com/308347/68534786-27942800-0341-11ea-8280-174bcc1dc18c.png)

So in the first case, the error span remains empty, while in the second case the warning and strike-through spans extend when typing.
